modules/build/src/main/scala/scala/build/options/ConfigMonoid.scala

```diff
@@ -81,17 +82,23 @@ object ConfigMonoid {
       main ++ defaults
   }
 
-  implicit def map[K, V](implicit valueMonoid: ConfigMonoid[V]): ConfigMonoid[Map[K, V]] =
+  implicit def map[K, V]: ConfigMonoid[Map[K, V]] =
     instance(Map.empty[K, V]) {
       (main, defaults) =>
         (main.keySet ++ defaults.keySet).map {
           key =>
-            val mainVal     = main.getOrElse(key, valueMonoid.zero)
-            val defaultsVal = defaults.getOrElse(key, valueMonoid.zero)
-            key -> valueMonoid.orElse(mainVal, defaultsVal)
+            val mainVal     = main.get(key)
+            val defaultsVal = defaults.get(key)
+            key -> mainVal.orElse(defaultsVal).get
         }.toMap
     }
 
+  implicit def stringOptionsList: ConfigMonoid[StringOptionsList] =
```

Comment:
This should live in the companion of `StringOptionsList` (we only put things that we can't change the companion of here).